Source/WebCore/Modules/compression/DecompressionStreamDecoder.cpp

```diff
@@ -118,7 +118,16 @@ ExceptionOr<RefPtr<JSC::ArrayBuffer>> DecompressionStreamDecoder::decompressZlib
     }
 
     while (shouldDecompress) {
-        auto output = Vector<uint8_t>(allocateSize);
+        if (allocateSize < startingAllocationSize)
+            return Exception { TypeError, "Out of Memory."_s };
+
+        auto output = Vector<uint8_t>();
+        if (!output.tryReserveInitialCapacity(allocateSize)) {
+            allocateSize /= 4;
+            continue;
```

Comment:
Here allocateSize is always initially equal to startingAllocationSize.
If the first tryReserveInitialCapacity fails, we will always return the exception.

Reply:
Exactly 😀.

We should have a minimum expectation of free memory that we can allocate for this operation. I think 16KB of memory is not unreasonable.

If you have a differing opinion on how to handle this please feel free to mention it.